src/viewport/viewport.service.ts

```diff
@@ -83,21 +84,22 @@ export class ViewportService {
 
 	constructor(
 		@Inject(UX_CONFIG) config: UxOptions,
+		@Inject(DOCUMENT) private document: any,
 		private windowRef: WindowRef,
 		private viewportServerSize: ViewportServerSizeService,
 	) {
 		if (windowRef.hasNative) {
 			this.resize$ = fromEvent<Event>(window, "resize").pipe(
 				map(() => this.getViewportSize()),
 				auditTime(config.viewport.resizePollingSpeed || UX_VIEWPORT_DEFAULT_CONFIG.resizePollingSpeed),
+				startWith(this.getViewportSize()),
```

Comment:
this was intendedly not here, as it will fire resize once (which we don't want) - it should be kept where it was for the sizetype

within `getViewportSize` add native check and return the SSR object there